File: my-folder/0128-longest-consecutive-sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> uniqueIntegers = Arrays.stream(nums).boxed().collect(Collectors.toSet());
        int maxCount=0;
        for (Integer element: uniqueIntegers) {
            int count=0;
            if(!uniqueIntegers.contains(element-1))
            {
                count++;
                int startElem=element+1;
                while(uniqueIntegers.contains(startElem))
                {
                  count++;
                  startElem++;
                }
                if(count>maxCount)
                    maxCount=count;
            }

        }

        return maxCount;
        
    }
}

File: my-folder/0128-longest-consecutive-sequence/solution.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        val numSet=nums.toHashSet()
        var lcs=0
        for(num in numSet)
        {
            if(!numSet.contains(num-1))
            {
                var cs=1;
                var next=num+1
                while(next in numSet)
                {
                  cs++
                  next++
                }
                if (lcs<cs) lcs=cs
            }
        }
        return lcs

    }
}
